scripts/plot_perts_x.jl

```diff
@@ -0,0 +1,194 @@
+                            bg,exp(x),𝕡)
+    #normalization for plotting, divide by integral of just momentum measure
+    ℳρ[i_x]=ℳρ[i_x] ./ bg.ρ₀ℳ(x) .* exp(-4x) #missing factor of a^-4 from pert rho
+end
+
+#I don't know how to do the DE.jl splining over all the perts at once so this is just an array...
+results_with_rsa = boltsolve_rsa(hierarchy; reltol=reltol)
+
+#Read in CLASS perturbations
+#CLASS keys (for reference):
+#['k (h/Mpc)', 'd_g', 'd_b', 'd_cdm', 'd_ur', 'd_ncdm[0]', 'd_tot',
+#'phi', 'psi', 't_g', 't_b', 't_cdm', 't_ur', 't_ncdm[0]', 't_tot']
+#reading files for two diff k modes
+# ret = open("./test/data/class_px_kp3.dat","r") do datafile
+ret = open("./test/data/class_px_kp03.dat","r") do datafile #note that these should only be used for comparing nonreion
+    [parse.(Float64, split(line)) for line in eachline(datafile)]
+end
+#By default CLASS uses fluid approximation, which introduces almost 2x error for massive neutrinos at lower x
+#So compare to no fluid case to see if hierarchy is right
+# retnf = open("./test/data/class_px_k1p0_nofluid_re.dat","r") do datafile
+# retnf = open("./test/data/class_px_kp3_nofluid.dat","r") do datafile
+retnf = open("./test/data/class_px_kp03_nofluid_re.dat","r") do datafile
+# an example that goes to early times -> retnf = open("./test/data/lowres_class_px_kp03_nofluid.dat","r") do datafile
+    [parse.(Float64, split(line)) for line in eachline(datafile)]
+end
+#the second column is just a repeated k value, so remember it and delete col
+kclass = retnf[2][1]
+class_pxs = transpose(reduce(hcat,ret[1:end .!= 2]))
+class_pxsnf = transpose(reduce(hcat,retnf[1:end .!= 2]))
+println("kclass is ", kclass, " kbolt is ",kbolt, " ratio (c/b) is ", kclass/kbolt)
+dipole_fac = k/bg.H₀/3e5*100/.7
+
+#Look at the massless species to check RSA
+
+#photon Θ0 monopole
+plot(class_pxs[1,:],log10.(abs.(class_pxsnf[2,:])),
+      label=raw"$\Theta_{0,\rm{CLASS}}$",legend=:topleft)
+plot!(x_grid, log10.(abs.(results_with_rsa[1,:]* 𝕡.h*4)),
+      label=raw"$4 h \Theta_{0,\rm{Bolt}}$",ls=:dash)
+vline!([this_rsa_switch],label="RSA switch",ls=:dot)
+xlims!(-8,0)
+xlabel!(raw"$x$")
+ylabel!(raw"$\delta_{i}(x)$")
+#photon Θ1 dipole
+plot(class_pxs[1,:],log10.(abs.(class_pxsnf[10,:])),
+     label=raw"$\Theta_{0,\rm{CLASS}}$",
+     legend=:topleft)
+plot!(x_grid, log10.(abs.(results_with_rsa[2,:] * dipole_fac)),
+      label=raw"$4 h \Theta_{0,\rm{Bolt}}$",ls=:dash)
+vline!([this_rsa_switch],label="RSA switch",ls=:dot)
+xlims!(-5,1)
+xlims!(-7,0)
+
+#neutrino 𝒩0 monopole
+plot(class_pxs[1,:],log10.(abs.(class_pxsnf[5,:])),
+      label=raw"$\mathcal{N}_{0,\rm{CLASS}}$",legend=:topleft)
+plot!(x_grid, log10.(abs.(results_with_rsa[1+2(ℓᵧ+1),:]* 𝕡.h*4)),
+      label=raw"$4 h \mathcal{N}_{0,\rm{Bolt}}$",ls=:dash)
+vline!([this_rsa_switch],label="RSA switch",ls=:dot)
+plot!(x_grid_rsa, log10.(abs.(4results_rsa[1+2(ℓᵧ+1),:]* 𝕡.h)),
```

Comment:
I think this should be `results_with_rsa`

Reply:
This line should've been removed - sorry! Should've been a little more careful with this script.